pkg/kv/store.go

```diff
@@ -0,0 +1,114 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+)
+
+var (
+	ErrClosedEntries       = errors.New("closed entries")
+	ErrConnectFailed       = errors.New("connect failed")
+	ErrDriverConfiguration = errors.New("driver configuration")
+	ErrMissingKey          = errors.New("missing key")
+	ErrMissingValue        = errors.New("missing value")
+	ErrNotFound            = errors.New("not found")
+	ErrOperationFailed     = errors.New("operation failed")
+	ErrPredicateFailed     = errors.New("predicate failed")
+	ErrSetupFailed         = errors.New("setup failed")
+	ErrUnknownDriver       = errors.New("unknown driver")
+)
+
+// Driver define the root interface to access the kv database.
+// Each kv provider implements a driver which provides Store instance.
+type Driver interface {
+	// Open access to the database store. Implementation should give access to the same storage based on the dsn.
```

Comment:
```suggestion
	// Open opens access to the database store. Implementations give access to the same storage based on the dsn.
```
(etc.) -- this should mostly be API documentation for kv users, *not* documentation for people who write Drivers.